src/header/mod.rs

```diff
@@ -110,27 +111,31 @@ pub struct Header {
 }
 
 impl Header {
-    pub(crate) fn parse(
+    pub(crate) const fn parse(
         input: &[u8],
     ) -> Result<(&[u8], (Header, Option<ColorTable<'_>>)), ParseError> {
         // File header
-        let (input, magic) = take::<2>(input)?;
-        if &magic != b"BM" {
+        let (input, magic) = propagate!(take2(input));
+
+        if let b"BM" = &magic {
+        } else {
             return Err(ParseError::InvalidFileSignature(magic));
         }
```

Comment:
Using `matches!` makes this slightly easier to read IMO:
```suggestion
        if !matches!(&magic, b"BM") {
            return Err(ParseError::InvalidFileSignature(magic));
        }
```